runtime/lua/vim/ui.lua

```diff
@@ -0,0 +1,32 @@
+local M = {}
+
+--- Prompts the user to pick a single item from a collection of entries
+---
+---@param entries table Arbitrary entries
+---@param opts table Additional options
+---     - prompt (string|nil)
+---               Text of the prompt. Defaults to `Pick one of:`
+---     - format_entry (function entry -> text)
+---               Function to format an
+---               individual entry from `entries`. Defaults to `tostring`.
+---@param callback function ((entry|nil, idx|nil) -> ())
+---               Called once the user made a choice.
+---               `idx` is the 1-based index of `entry` within `entries`.
+---               `nil` if the user aborted the dialog.
+function M.pick_one(entries, opts, callback)
```

Comment:
maybe we can find a name that (1) draws on existing conventions and (2) generalizes beyond the "picker" case. "pick" is a new verb that comes from vscode I assume. 

Since this appears to be a thin wrapper around input(),  consider naming this ...
- `vim.ui.input()` for discoverability and analogy with the existing `input()` (also common in [HTML forms](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input))
- some other verb that's common in UI frameworks. Typically "select" (also in [HTML](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select))

"input" seems like it works. We could have `vim.ui.input()`, `vim.ui.input_multi()`, and other ``vim.ui.input_xx()` if we need more advanced things.

Reply:
I think `input` is ambiguous - `vim.fn.input` already exists and is for taking a literal string as input.

I'd rather go with `select` in that case, though picker / pick is also used in a couple of UI frameworks - we're not inventing a new term, but I'm fine with either.

Reply:
> `vim.fn.input` already exists and is for taking a literal string as input.

Please do _not_ choose names that are different from existing vimscript functions for that purpose. That is completely the wrong practice. It results in a mismash of APIs that are similar but have N different names for the same concept.

> though picker / pick is also used in a couple of UI frameworks - we're not inventing a new term, 

It is new in vim ecosystem.

Reply:
> It results in a mismash of APIs that are similar but have N different names for the same concept.

Prompting a user for an input string (`vim.fn.input`) and asking the user to select a single element from a collection of elements (`vim.ui.pick_one`) are two different concepts. Yes, you can zoom out and view both as "asking user for input", but in the context of how these two functions are going to be used, the different semantic is important.

To be clear, I'm not hung up on the `pick_one` name. But I'm pretty sure that using `input` is not the best idea idea either.

If we wanted to stick with existing naming, I'd rather go with another `inputlist`, though I personally dislike the term because it is focused on what is supplied as main parameter, and not on what the function does. (And speaking of existing functions, there is also `confirm` which is also kinda similar).

Reply:
+1 for `select`. It's short, simple, easy to understand, maps well to this use case, and has existing usage with similar semantics (e.g. in HTML).